parser/src/java/org/apache/hadoop/hive/ql/parse/AlterClauseParser.g

```diff
@@ -460,11 +468,16 @@ compactPool
   -> ^(TOK_COMPACT_POOL $poolName)
   ;
 
+fileSizeThresholdClause
+  : KW_FILE_SIZE_THRESHOLD EQUAL value=StringLiteral 
+  -> ^(TOK_FILE_SIZE_THRESHOLD $value)
+  ;
+
 alterStatementSuffixCompact
 @init { gParent.msgs.push("compaction request"); }
 @after { gParent.msgs.pop(); }
-    : KW_COMPACT compactType=StringLiteral tableImplBuckets? blocking? compactPool? (KW_WITH KW_OVERWRITE KW_TBLPROPERTIES tableProperties)? orderByClause? whereClause?
-    -> ^(TOK_ALTERTABLE_COMPACT $compactType tableImplBuckets? blocking? compactPool? tableProperties? orderByClause? whereClause?)
+    : KW_COMPACT compactType=StringLiteral tableImplBuckets? blocking? fileSizeThresholdClause? compactPool? (KW_WITH KW_OVERWRITE KW_TBLPROPERTIES tableProperties)? whereClause? orderByClause?
+    -> ^(TOK_ALTERTABLE_COMPACT $compactType tableImplBuckets? blocking? fileSizeThresholdClause? compactPool? tableProperties? whereClause? orderByClause?)
```

Comment:
i don't think we should supply table properties via compaction command, we should set it on table level

Reply:
Done